### PR TITLE
HOSTEDCP-676 CAPI AWS provider pod from latest 4.13 release payload crashloops continuously

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
Fixes the issue where the CAPI AWS provider pod from a 4.13 release payload crashloops continuously by bumping the CAPI AWS version to v2.0.2 (including dependencies).

**Which issue(s) this PR fixes**:
Fixes [HOSTEDCP-676](https://issues.redhat.com/browse/HOSTEDCP-676)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by bryan-cox** [comment]
Rerunning unit tests since prow didn't pick up the changes I just pushed.

/test unit

**Comment by bryan-cox** [comment]
Rerunning AWS tests since those seem to be working again in CI.
/test e2e-aws

File: hypershift-operator/controllers/nodepool/aws.go [hunk 1]
```diff
@@ -24,7 +24,6 @@ func awsMachineTemplateSpec(infraName, ami string, hostedCluster *hyperv1.Hosted
 	subnet := &capiaws.AWSResourceReference{}
 	if nodePool.Spec.Platform.AWS.Subnet != nil {
 		subnet.ID = nodePool.Spec.Platform.AWS.Subnet.ID
-		subnet.ARN = nodePool.Spec.Platform.AWS.Subnet.ARN
```
Comment (enxebre) [inline hypershift-operator/controllers/nodepool/aws.go:27]: we'll need to align here as well https://github.com/openshift/hypershift/blob/main/api/v1beta1/nodepool_types.go#L628
That can be a follow up.

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go [hunk 2]
```diff
@@ -984,84 +984,6 @@ func (c *createTypeTrackingClient) Create(ctx context.Context, obj crclient.Obje
 	return c.Client.Create(ctx, obj, opts...)
 }
 
-func TestReconcileAWSSubnets(t *testing.T) {
-	g := NewGomegaWithT(t)
```
[the remark this answers is not in the document]
Comment (enxebre) [reply hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go:988]: nvm got it

**Comment by enxebre** [comment]
I'm not seeing where this PR is removing the capa harcoded override https://github.com/openshift/hypershift/pull/1933/files#diff-9132b8504bc00c929d8a04a3f9235d08a8489ce1effe9fa78eafcc1d7acc7c8fR239 ?

**Comment by enxebre** [comment]
This is looking great, can we validate this works with 4.12 and 4.13 after addressing https://github.com/openshift/hypershift/pull/2000#issuecomment-1385065122?

Is the CRD definition for capa resources using?
  conversion:
    strategy: None

**Comment by bryan-cox** [comment]
> @bryan-cox I'm not seeing where this PR is removing the capa harcoded override https://github.com/openshift/hypershift/pull/1933/files#diff-9132b8504bc00c929d8a04a3f9235d08a8489ce1effe9fa78eafcc1d7acc7c8fR239 ?

@enxebre, you should be able to find that [here](https://github.com/openshift/hypershift/pull/2000/files#diff-9132b8504bc00c929d8a04a3f9235d08a8489ce1effe9fa78eafcc1d7acc7c8f).

File: hypershift-operator/controllers/nodepool/aws.go (hunk 1 above)
Comment (bryan-cox) [reply hypershift-operator/controllers/nodepool/aws.go:27]: Thanks for catching that! I went ahead and removed it and added the changes to the `Remove ARN from AWS machine template generation` [commit](https://github.com/openshift/hypershift/pull/2000/commits/cec09b724709a022177faf7de743dc5390373a91).

**Comment by bryan-cox** [comment]
> This is looking great, can we validate this works with 4.12 and 4.13 after addressing [#2000 (comment)](https://github.com/openshift/hypershift/pull/2000#issuecomment-1385065122)?
> 
> Is the CRD definition for capa resources using? conversion: strategy: None

@enxebre I'm assuming you mean for the AWSMachines? If so, yeah:
```
[bcox@localhost hypershift]$ oc get crds/awsmachines.infrastructure.cluster.x-k8s.io -o yaml | grep "conversion:" -A 2
  conversion:
    strategy: None
  group: infrastructure.cluster.x-k8s.io
```

Also, I previously verified this worked with both 4.12 and 4.13. I retested since I made a change to the commit [Remove ARN from AWS machine template generation](https://github.com/openshift/hypershift/pull/2000/commits/cec09b724709a022177faf7de743dc5390373a91) from this [comment](https://github.com/openshift/hypershift/pull/2000#discussion_r1071921958). It still works with both 4.12 and 4.13:
```
[bcox@localhost hypershift]$ ./bin/hypershift create cluster aws --name $CLUSTER_NAME-412 --node-pool-replicas=2 --base-domain $BASE_DOMAIN --pull-secret $PULL_SECRET --aws-creds $AWS_CREDS --region $REGION --generate-ssh --release-image registry.ci.openshift.org/ocp/release:4.12.0-0.ci-2023-01-17-075213
[bcox@localhost hypershift]$ ./bin/hypershift create cluster aws --name $CLUSTER_NAME-413-nightly --node-pool-replicas=2 --base-domain $BASE_DOMAIN --pull-secret $PULL_SECRET --aws-creds $AWS_CREDS --region $REGION --generate-ssh --release-image registry.ci.openshift.org/ocp/release:4.13.0-0.nightly-2023-01-11-225449
```
```
[bcox@localhost hypershift]$ oc get nodepools --namespace clusters
NAME                                          CLUSTER                            DESIRED NODES   CURRENT NODES   AUTOSCALING   AUTOREPAIR   VERSION                              UPDATINGVERSION   UPDATINGCONFIG   MESSAGE
brcox-hypershift-x86-412-us-east-1a           brcox-hypershift-x86-412           2               2               False         False        4.12.0-0.ci-2023-01-17-075213
brcox-hypershift-x86-413-nightly-us-east-1a   brcox-hypershift-x86-413-nightly   2               2               False         False        4.13.0-0.nightly-2023-01-11-225449

[bcox@localhost hypershift]$ oc get --namespace clusters hostedclusters
NAME                               VERSION                              KUBECONFIG                                          PROGRESS    AVAILABLE   PROGRESSING   MESSAGE
brcox-hypershift-x86-412           4.12.0-0.ci-2023-01-17-075213        brcox-hypershift-x86-412-admin-kubeconfig           Completed   True        False         The hosted control plane is available
brcox-hypershift-x86-413-nightly   4.13.0-0.nightly-2023-01-11-225449   brcox-hypershift-x86-413-nightly-admin-kubeconfig   Completed   True        False         The hosted control plane is available
```

**Comment by enxebre** [comment]
>@bryan-cox I'm not seeing where this PR is removing the capa harcoded override https://github.com/openshift/hypershift/pull/1933/files#diff-9132b8504bc00c929d8a04a3f9235d08a8489ce1effe9fa78eafcc1d7acc7c8fR239 ?
@enxebre, you should be able to find that [here](https://github.com/openshift/hypershift/pull/2000/files#diff-9132b8504bc00c929d8a04a3f9235d08a8489ce1effe9fa78eafcc1d7acc7c8f).

Ah I see, that's sneaking in the bumping/generation commit. Can we pease move that change into a different commit?

**Comment by bryan-cox** [comment]
> > @bryan-cox I'm not seeing where this PR is removing the capa harcoded override https://github.com/openshift/hypershift/pull/1933/files#diff-9132b8504bc00c929d8a04a3f9235d08a8489ce1effe9fa78eafcc1d7acc7c8fR239 ?
> > @enxebre, you should be able to find that [here](https://github.com/openshift/hypershift/pull/2000/files#diff-9132b8504bc00c929d8a04a3f9235d08a8489ce1effe9fa78eafcc1d7acc7c8f).
> 
> Ah I see, that's sneaking in the bumping/generation commit. Can we pease move that change into a different commit?

I pulled it out and moved it to its own commit.

**Comment by enxebre** [comment]
for @csrwng / @sjenning awareness to take a final look

**Pull request merged**